### PR TITLE
Add error checking functions

### PR DESCRIPTION
**Pull request by Craftsman8**
#  The new features that were added are as follows:

- Add a conditional error message to display on the UI.
- Use setTimeout() function to delay the server for 5s before loading page.
- Add a loading message on the UI to indicate running process.

### CODE REVIEW

File: src/Components.js [hunk 1]
```diff
@@ -3,19 +3,32 @@ import DeleteButton from "./DeleteButton";
...
+            .catch((error)=>{
+                console.log(error.message)
+                setError(error.message)
+                setLoading(false)
+            })
+        }, 5000)
+    }, []); //empty dependency array - allows the app to run only once
 
     const handleDelete = (id) => {
         console.log('The id of the clicked component is', id)
```
Comment (Bestbynature) [inline src/Components.js:34]: Hello @Craftsman8.

Weldone on adding the features requested in this project. You are making a significant progress. Could you remove the delete functionality as we are not using it right now? 

Just so you know, removing the delete functionality will break the app if you don't remove all the references to that particular function immediately after removing the delete function.

Comment (Bestbynature) [reply src/Components.js:34]: Kindly remove the unused students' array in the db.json file. as it is not needed at the moment.

Comment (Bestbynature) [reply src/Components.js:34]: Kindly ensure that your app is running before you resubmit the link as it is not running currently. It is breaking.

**Review by Bestbynature — CHANGES_REQUESTED** [review]
Good job so far 💯 

# Highlights
- Good commit messages
- on-time submission
- intuitive error messages


## You are almost there. 

Almost There GIF

Please check the comments I left on your project and fix each point. 


Once you are done with the corrections given in the review, please follow the steps below to resubmit:

- [x] git add .
- [x] git commit -m "your commit message"
- [x] git push
- [x] Don't create another pull request. **Just resubmit this same link**

File: src/Components.js (hunk 1 above)
Comment (Craftsman8) [reply src/Components.js:34]: I'll do as you've instructed sir.. 🙇
I actually added an error to the server link so I can see my error message, but I forgot to remove it before submission

**Pull request merged**